firmware/Sources/HAL/HAL_PWM/hal_pwm.c

```diff
@@ -0,0 +1,104 @@
+/*                     Definition of local symbolic constants                     */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                    Definition of local function like macros                    */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*            Definition of local types (typedef, enum, struct, union)            */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                         Definition of local variables                          */
+/**********************************************************************************/
+static HRTIM_CompareCfgTypeDef pCompareCfg = {0};
+/**********************************************************************************/
+/*                        Definition of exported variables                        */
+/**********************************************************************************/
+extern HRTIM_HandleTypeDef hhrtim1;
+
+/**********************************************************************************/
+/*                      Definition of exported constant data                      */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                    Declaration of local function prototypes                    */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                       Definition of local constant data                        */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                         Definition of local functions                          */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                        Definition of exported functions                        */
+/**********************************************************************************/
+
+HAL_PWM_result_e HAL_PwmInit(void){
+	HAL_PWM_result_e res = HAL_PWM_RESULT_SUCCESS;
+	EPC_ST_ERR_COUNTER = 0;
+	MX_HRTIM1_Init();
+	if (EPC_ST_ERR_COUNTER){
+		res = HAL_PWM_RESULT_ERROR;
+	}
+	return res;
+}
+
+
+HAL_PWM_result_e HAL_PwmSetDuty(const uint32_t duty){
+	uint32_t max = hhrtim1.Instance->sTimerxRegs[0].PERxR;
+	uint32_t pwm_duty;
+	if (duty>100000){
```

Comment:
no magic numbers. 

add internal #define to set constants

Reply:
Done